core/db_classes/EE_Transaction.class.php

```diff
@@ -1403,4 +1700,29 @@ public function verify_abandoned_transaction_status()
             }
         }
     }
+
+
+    /**
+     * @param string $source function name that called this method
+     * @return boolean | int
+     * @throws EE_Error
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidInterfaceException
+     * @throws ReflectionException
+     * @throws RuntimeException
+     */
+    public function delete($source = 'unknown')
+    {
+        $current_user = wp_get_current_user();
+        $this->add_extra_meta(
+            EE_Transaction::EXTRA_META_KEY_TXN_DELETED,
+            array(
+                'deleted-by' => $current_user->ID ? $current_user->display_name : 'unauthed user',
+                'timestamp'  => time(),
+                'source'     => $source,
+            )
+        );
+        return parent::delete();
+    }
```

Comment:
new method

Reply:
As previously mentioned in `EE_Registration::delete` methods, I think it'd be better to utilize actions for this instead of always logging.  At a minimum, the extra logging should be controlled via either a constant or an option.